indico/htdocs/js/indico/jquery/categorypicker.js

```diff
@@ -0,0 +1,332 @@
+            }).attr('autocomplete', 'off');
+
+            // Insert elements in DOM
+            var $typeaheadForm = $($.parseHTML(self._typeaheadTemplate)[0]);
+            $typeaheadForm.find('.typeahead__query').append(self.$searchInput);
+            self.element.prepend($typeaheadForm);
+
+            // Typeahead init
+            self.$searchInput.typeahead({
+                dynamic: true,
+                filter: false,
+                source: {
+                    ajax: {
+                        url: build_url(Indico.Urls.Categories.search),
+                        path: 'categories',
+                        data: {
+                            q: "{{query}}"
+                        }
+                    }
+                },
+                resultContainer: false,
+                cancelButton: false,
+                display: 'title',
+                callback: {
+                    onClickAfter: function(node, a, item) {
+                        self.goToCategory(item.id);
+                    },
+                    onResult: function(node, query, result) {
+                        self._renderResults(result);
+                    },
+                    onHideLayout: function() {
+                        self.$categoryResults.html('');
+                        self.$category.show();
+                        self.$categoryTree.show();
+                    },
+                    onShowLayout: function(node, query) {
+                        self.$category.hide();
+                        self.$categoryTree.hide();
+                    }
+                }
+            });
+        },
+
+        _buildBreadcrumbs: function(path, clickable) {
+            var self = this;
+            var $breadcrumbs = $('<ul>', {class: 'breadcrumbs'});
+            var tag = clickable? '<a>' : '<span>';
+
+            _.each(path, function(category) {
+                var $item = $(tag, {
+                    text: category.title
+                });
+                if (clickable) {
+                    $item.attr('href', '');
+                    $item.attr('title', $T.gettext("Go to: {0}".format(category.title)));
+                }
+                self._bindGoToCategoryOnClick($item, category.id);
+                $breadcrumbs.append($('<li>').append($item));
+            });
+
+            return $breadcrumbs;
+        },
+
+        _buildCategory: function(category, isSubcategory, withBreadcrums, clickableBreadcrumbs) {
+            var self = this;
+            var tag = isSubcategory? '<li>' : '<div>';
+            var itemClass = isSubcategory? 'subcategory' : 'current-category';
+
+            var $category = $(tag, {
+                class: 'item ' + itemClass,
+                id: 'category-' + category.id
+            });
+
+            var $categoryTitle = $('<div>', {class: 'title-wrapper'});
+            $categoryTitle.append($('<span>', {class: 'title',text: category.title}));
```

Comment:
When there are more than 1 elements in the `dict` I prefer to put them in a new line.